src/scripts/search.js

```diff
@@ -0,0 +1,134 @@
+/******************************************************************************
+
+Flatmap viewer and annotation tool
+
+Copyright (c) 2019  David Brooks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+******************************************************************************/
+
+
+//==============================================================================
+
+import MiniSearch from 'minisearch';
+
+//==============================================================================
+
+// The properties of a feature we index and show
+
+export const indexedProperties = [
+    'label',
+    'models',
+    'source'
+];
+
+//==============================================================================
+
+export class SearchIndex
+{
+    constructor()
```

Comment:
Using constructor and class may cause problem on mapintegratedvuer.

Reply:
Should I switch to function?